src/module/actor/lancer-actor.ts

```diff
@@ -25,13 +25,42 @@ import { LancerHooks, LancerSubscription } from "../helpers/hooks";
 import { mm_wrap_actor } from "../mm-util/helpers";
 import { system_ready } from "../../lancer";
 import type { LancerItemType } from "../item/lancer-item";
-import { renderMacroTemplate, encodeMacroData } from "../macros";
+import { renderMacroTemplate, prepareTextMacro, encodeMacroData } from "../macros";
 import type { RegEntry, MechWeapon, NpcFeature } from "machine-mind";
 import { StabOptions1, StabOptions2 } from "../enums";
 import { fix_modify_token_attribute } from "../token";
 import type { ActionData } from "../action";
 const lp = LANCER.log_prefix;
 
+export function prepareStructureSecondaryRollMacro(registryId: string) {
+  // @ts-ignore
+  let roll = new Roll('1d6').evaluate({ async: false });
+  let result = roll.total!;
+  if (result <= 3) {
+    prepareTextMacro(registryId, "Destroy Weapons", `
```

Comment:
We should make a way for this function to pass the `roll` to the `ChatMessage`, so that modules that interact with new rolls (f.e. Dice So Nice) can catch it. I don't think we have a generic `prepareRollMacro` though...

Reply:
yep, fully agreed. This is all just minimum viable touching of this code while I was getting rid of `eval` in macros.